### DISCUSSION TITLE
How do I offload all custom element definitions into a different js file and then import it

### DISCUSSION THREAD

**Discussion by IshaanShettigar** [clientIO/joint#2127] [Q&A] [answered]
### Introduction

Sorry, this seems like a noob-level question but I'm working with many custom elements and everything is working great. The problem is I need to use these custom elements on different pages so I thought it'd be great if I offloaded the custom element definitions into its own js file and then just imported it wherever necessary. This can be done for many things like LinkTools, ElementTools etc. Currently, I have defined everything wherever I'm using it, if I were to separate it then that would save me a lot of lines of code. 

I had imported the script in my HTML as type="text/javascript" then changed it to "module" because I got an error "Can't use import outside a module". I then got a CORS error, so I installed express and made a server.js which I ran using `node server.js`.
Then it seemed like everything was working fine but my elementools no longer loaded onto my elements for some reason, I removed the type="module" from my script tag in HTML and voila it worked again, I do not know why. (Keep in mind I still haven't imported anything)

**To sum up, adding` type="module" `to my script tag seems to break the joint js code, but why and is there a workaround**

### Steps to reproduce

Repo link:  https://github.com/IshaanShettigar/iDG10-GUI-Hackathon
clone it and run node server.js
navigate to localhost:3000/mainDemo.html

To reproduce the error, go to mainDemo.html and add `type="module"` in the script tag where I have imported mainDemo.js.  double click on an element on the left-hand side, then go to the console to see the error NewEleView not defined, why does this happen? 
Note: This doesn't happen if I remove the type="module"

I do not know what I am missing. Someone, please help

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by jamesgeorgewilliams** [comment, marked as answer]
Hi,

The following demo shows a simple _JointJS_ diagram with some custom elements in a separate file `shapes.js`. If it is necessary for your project to separate code, you can follow a similar structure using JS modules and `import` and `export` syntax.

https://codesandbox.io/s/jointjs-paper-shapes-u9vxdt

It's difficult to comment on your project specifically as I am not sure how you tried to separate the files originally, but if you have any trouble with the above approach, you can create a similar codesandbox example, and we can take a look. Thanks
